examples/client/Locomotion/src/Components/AvailabilityVehicle/index.tsx

```diff
@@ -42,34 +52,34 @@ const AvailabilityVehicle = ({
 }: AvailabilityVehicleProps) => {
   const { useVehicleColor } = useContext(ThemeContext);
   const { vehicleColor } = useVehicleColor();
-  const markerRef = useRef<MarkerAnimated>(null);
-  const [locationAnimated] = useState(new AnimatedRegion({
-    latitude: insureNumberType(location.lat),
-    latitudeDelta: 0.1,
-    longitude: insureNumberType(location.lng),
-    longitudeDelta: 0.1,
-  }));
+  const markerRef = useRef<MapMarker>(null);
+  const locationAnimationRef = useRef<AnimatedRegion>(
+    new AnimatedRegion({
+      latitude: ensureNumberType(location.lat),
+      longitude: ensureNumberType(location.lng),
+      latitudeDelta: MAINTAIN_CURRENT_ZOOM,
+      longitudeDelta: MAINTAIN_CURRENT_ZOOM,
+    }),
+  );
 
   useEffect(() => {
     if (location.lat && location.lng) {
-      if (Platform.OS === 'android') {
-        setTimeout(() => {
-          markerRef?.current?.animateMarkerToCoordinate({
-            latitude: insureNumberType(location.lat),
-            longitude: insureNumberType(location.lng),
-          }, DURATION);
-        }, 0);
-      } else {
-        locationAnimated.timing({
-          latitude: location.lat,
-          longitude: location.lng,
-          useNativeDriver: false,
-          duration: DURATION,
-        }).start();
-      }
+      locationAnimationRef.current.timing({
+        toValue: ANIMATED_VALUE_PLACEHOLDER,
+        duration: DURATION,
+        longitudeDelta: MAINTAIN_CURRENT_ZOOM,
+        latitudeDelta: MAINTAIN_CURRENT_ZOOM,
+        useNativeDriver: false,
+        latitude: location.lat,
+        longitude: location.lng,
+      }).start();
     }
   }, [location]);
```

Comment:
can be
```suggestion
  }, [location.lat, location.lng]);
```
or we need other location props?

Reply:
Technically, "it doesn't matter" (it does matter because using an object as a useEffect dep is generally wrong), because the `location` object reference changes every time we fetch the vehicles from the API, so this works.
Changed to what you suggested because it makes more sense - but the correct solution here would be to pass `...location`, aka `lat`, `long` and `bearing` instead of the entire object. Sadly, I don't have time for that ATM

Reply:
oki doki